curriculum/challenges/english/07-scientific-computing-with-python/learn-interfaces-by-building-an-equation-solver/664eef158d792a509e8d708a.md

```diff
@@ -0,0 +1,115 @@
+---
+id: 664eef158d792a509e8d708a
+title: Step 37
+challengeType: 20
+dashedName: step-37
+---
+
+# --description--
+
+The `sub` function from the `re` module enables you to replace text inside a string based on a regex pattern.
+
+```py
+spam = 'Always look on the bright side of life'
+ham = re.sub('bright', 'light', spam)
+ham == 'Always look on the light side of life' # True
+```
+
+It takes three arguments: the regex pattern to match, the replacement, and the string on which you want to perform the replacement.
+
+From your `__str__` function, return a `sub()` call passing the string `'1'`, an empty string, and your existing `equation_string.strip('+')` call as the arguments. This will replace each `1` with an empty string. Don't worry, as you'll refine the regex pattern in the next steps.
```

Comment:
```suggestion
From your `__str__` function, return a `sub()` call passing the string `'1'`, an empty string, and your existing `equation_string.strip('+')` call as the arguments. This will replace each `1` with an empty string. The result is not refined yet and you'll continue to work on the regex pattern in the next steps.
```